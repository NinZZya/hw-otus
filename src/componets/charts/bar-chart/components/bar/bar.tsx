import React from "react";
import { Rect, Text } from "react-konva";
/** @jsx jsx */
import { jsx } from "@emotion/core";

export interface BarProp {
  x: number;
  y: number;
  width: number;
  height: number;
  value: number;
  label: string;
  margin: number;
  padding: number;
  barColor: string;
  textColor: string;
}

const DefaultColor = {
  BAR: "#3700b3",
  TEXT: "#ffffff",
};

export const Bar: React.FC<BarProp> = (props) => {
  const { x, y, width, height, value, label, margin, padding } = props;
  const valueY = y - margin - padding;
  const labelY = y + height + padding;
  const barColor = props.barColor ? props.barColor : DefaultColor.BAR;
  const textColor = props.textColor ? props.textColor : DefaultColor.TEXT;

  return (
    <>
      <Text text={String(value)} fill={textColor} x={x} y={valueY} />
      <Rect x={x} y={y} width={width} height={height} fill={barColor} />
      <Text text={label} fill={textColor} x={x} y={labelY} />
    </>
  );
};
