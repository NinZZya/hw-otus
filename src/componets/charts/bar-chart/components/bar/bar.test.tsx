import React from "react";
import renderer from "react-test-renderer";
import { Bar } from "./bar";

const bar = {
  x: 10,
  y: 20,
  width: 30,
  height: 100,
  value: 100,
  label: "Label",
  margin: 10,
  padding: 10,
  barColor: "#ffffff",
  textColor: "#0000000",
};

describe(`Bar render coorectly`, () => {
  it(`Bar with data render coorectly`, () => {
    const tree = renderer.create(<Bar {...bar} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
