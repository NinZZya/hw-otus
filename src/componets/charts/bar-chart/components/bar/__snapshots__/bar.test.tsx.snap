// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar render coorectly Bar with data render coorectly 1`] = `
Array [
  <Text
    fill="#0000000"
    text="100"
    x={10}
    y={0}
  />,
  <Rect
    fill="#ffffff"
    height={100}
    width={30}
    x={10}
    y={20}
  />,
  <Text
    fill="#0000000"
    text="Label"
    x={10}
    y={130}
  />,
]
`;
